tests/test_agents.py

```diff
@@ -0,0 +1,976 @@
+from copy import copy, deepcopy
+
+import pandas as pd
+import polars as pl
+import pytest
+
+from mesa_frames import AgentsDF, ModelDF
+from mesa_frames.abstract.agents import AgentSetDF
+from mesa_frames.types import MaskLike
+from tests.test_agentset_pandas import (
+    ExampleAgentSetPandas,
+    fix1_AgentSetPandas,
+    fix2_AgentSetPandas,
+)
+from tests.test_agentset_polars import (
+    ExampleAgentSetPolars,
+    fix2_AgentSetPolars,
+)
+
+
+# This serves otherwise ruff complains about the two fixtures not being used
+def not_called():
+    fix1_AgentSetPandas()
+    fix2_AgentSetPandas()
+    fix2_AgentSetPolars()
+
+
+@pytest.fixture
+def fix_AgentsDF(
+    fix1_AgentSetPandas: ExampleAgentSetPandas,
+    fix2_AgentSetPolars: ExampleAgentSetPolars,
+) -> AgentsDF:
+    model = ModelDF()
+    agents = AgentsDF(model)
+    agents.add([fix1_AgentSetPandas, fix2_AgentSetPolars])
+    return agents
+
+
+class Test_AgentsDF:
+    def test___init__(self):
+        model = ModelDF()
+        agents = AgentsDF(model)
+        assert agents.model == model
+        assert isinstance(agents._agentsets, list)
+        assert len(agents._agentsets) == 0
+        assert isinstance(agents._ids, pl.Series)
+        assert agents._ids.is_empty()
+        assert agents._ids.name == "unique_id"
+
+    def test_add(
+        self,
+        fix1_AgentSetPandas: ExampleAgentSetPandas,
+        fix2_AgentSetPolars: ExampleAgentSetPolars,
+    ):
+        model = ModelDF()
+        agents = AgentsDF(model)
+        agentset_pandas = fix1_AgentSetPandas
+        agentset_polars = fix2_AgentSetPolars
+
+        # Test with a single AgentSetPandas
+        result = agents.add(agentset_pandas, inplace=False)
+        assert result._agentsets[0] is agentset_pandas
```

Comment:
You should only test via the public API's, because private structure may change all the time.

Reply:
You're right, I thought that the public API would be more unstable, but as we continue development, it is the other way around. I will open an issue to change tests from private to public APIs in the future.